src/components/Table/utils/columns.tsx

```diff
@@ -163,15 +168,20 @@ export function assignDefaultColumnIds<T extends Kinded>(columns: GridColumn<T>[
   // Note: we are not _always_ spreading the `c` property as we need to be able to return the whole proxy object that
   // exists as part of `selectColumn` and `collapseColumn`.
   return columns.map((c, idx) => {
-    const { expandColumns = [] } = c;
-    const expandColumnsWithId: GridColumnWithId<T>[] | undefined = expandColumns.map((ec, ecIdx) => ({
-      ...ec,
-      id: ec.id ?? (`${generateColumnId(idx)}_${ecIdx}` as string),
-      // Defining this as undefined to make TS happy for now.
-      // If we do not explicitly set to `undefined`, TS thinks `expandColumns` could still be of type GridColumn<T> (not WithId).
-      // We only support a single level of expanding columns, so this is safe to do.
-      expandColumns: undefined,
-    }));
+    const { expandColumns } = c;
+    const expandColumnsWithId: GridColumnWithId<T>[] | number | string | undefined =
+      expandColumns === undefined
+        ? undefined
+        : Array.isArray(expandColumns)
+        ? expandColumns.map((ec, ecIdx) => ({
+            ...ec,
+            id: ec.id ?? (`${generateColumnId(idx)}_${ecIdx}` as string),
+            // Defining this as undefined to make TS happy for now.
+            // If we do not explicitly set to `undefined`, TS thinks `expandColumns` could still be of type GridColumn<T> (not WithId).
+            // We only support a single level of expanding columns, so this is safe to do.
+            expandColumns: undefined,
+          }))
+        : expandColumns;
```

Comment:
May be able to just simplify to `!Array.isArray(expandColumns) ? expandColumns : expandColumns.map(...)`